base_classes/NXcg_primitive.nxdl.xml

```diff
@@ -0,0 +1,229 @@
+    </doc>
+    <field name="depends_on" type="NX_CHAR">
+        <doc>
+             Reference to an instance of :ref:`NXcoordinate_system` in which these primitives
+             are defined.
+        </doc>
+    </field>
+    <field name="dimensionality" type="NX_POSINT" units="NX_UNITLESS">
+        <doc>
+             The dimensionality of the primitive set.
+        </doc>
+        <enumeration>
+            <item value="1"/>
+            <item value="2"/>
+            <item value="3"/>
+        </enumeration>
+    </field>
+    <field name="cardinality" type="NX_POSINT" units="NX_UNITLESS">
+        <doc>
+             The cardinality of the primitive set.
+        </doc>
+    </field>
+    <field name="identifier_offset" type="NX_INT" units="NX_UNITLESS">
+        <doc>
+             Integer offset whereby the identifier of the first member
+             of the set differs from zero.
+             
+             Identifiers can be defined either implicitly or explicitly.
+             For implicit indexing identifiers are defined on the interval
+             :math:`[identifier\_offset, identifier\_offset + c - 1]`.
+             
+             Therefore, implicit identifier are completely defined by the value of
+             identifier_offset and cardinality. For example if identifier run from
+             -2 to 3 the value for identifier_offset is -2.
+             
+             For explicit indexing the field identifier has to be used.
+             Fortran-/Matlab- and C-/Python-style indexing have specific implicit
+             identifier conventions where identifier_offset is 1 and 0 respectively.
+        </doc>
+    </field>
+    <field name="identifier" type="NX_INT">
+        <doc>
+             Identifier of each member for explicit indexing.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="center" type="NX_NUMBER" units="NX_ANY">
+        <doc>
+             The center of mass position of each primitive.
+        </doc>
+        <dimensions rank="2">
+            <dim index="1" value="c"/>
+            <dim index="2" value="d"/>
+        </dimensions>
+    </field>
+    <field name="is_center_of_mass" type="NX_BOOLEAN">
+        <doc>
+             True if the center is a center of mass.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="shape" type="NX_NUMBER" units="NX_LENGTH">
+        <doc>
+             A qualitative description of the shape of each primitive.
+        </doc>
+        <dimensions rank="2">
+            <dim index="1" value="c"/>
+            <dim index="2" value="d"/>
+        </dimensions>
+    </field>
+    <field name="length" type="NX_NUMBER" units="NX_LENGTH">
+        <doc>
+             Qualifier for the length of characteristic features of the primitive.
```

Comment:
How does a number qualify a length?

Reply:
similar though in my head when drafting it but agreed that qualitative is not the right word at the point when I drafted it I was comparing always to people reporting vertices and faces (i.e. full geometry not just scalar properties).